### PR TITLE
:hammer::art::microscope: Refactor to improve code maintainability

### PR DESCRIPTION
**Pull request by bbody**

### CODE REVIEW

**Comment by bbody** [comment]
Includes:
- https://github.com/bbody/CMD-Resume/issues/235
- https://github.com/bbody/CMD-Resume/issues/84

File: js/helpers/misc.js [hunk 1]
```diff
@@ -21,7 +21,42 @@ var isValidColor = function(color) {
 	}
 };
 
+// Get value
+var hasArrayKey = function(data, keys) {
+	for (var i = 0; i < keys.length; i++) {
```
Comment (obelic2) [inline js/helpers/misc.js]: Why not a forEach? Would read better and reduce lines

File: js/helpers/misc.js [hunk 2]
```diff
@@ -21,7 +21,42 @@ var isValidColor = function(color) {
...
+	}
+
+	var data = hasArrayKey(object, getKeyArray(key));
+
+	return data && (!!isObject || data.length);
+};
+
+// Get data from object
+var getData = function(object, keys) {
+	return hasArrayKey(object, Array.isArray(keys) ? keys : [keys]);
```
Comment (obelic2) [inline js/helpers/misc.js]: You already have define a getKeyArray function

File: js/helpers/misc.js (hunk 1 above)
Comment (bbody) [reply js/helpers/misc.js]: Because then I would have to polyfill

File: js/helpers/misc.js (hunk 2 above)
Comment (bbody) [reply js/helpers/misc.js]: Fixed

**Pull request merged**